configs/configupgrade/analysis.go

```diff
@@ -241,11 +241,11 @@ func (u *Upgrader) analyze(ms ModuleSources) (*analysis, error) {
 		return nil, fmt.Errorf("error resolving providers:\n%s", errorsMsg)
 	}
 
-	for name, fn := range providerFactories {
-		log.Printf("[TRACE] Fetching schema from provider %q", name)
+	for fqn, fn := range providerFactories {
+		log.Printf("[TRACE] Fetching schema from provider %q", fqn.LegacyString())
```

Comment:
This is not a big deal at all, and certainly not worth doing a bulk change everywhere for it yet, but I just wanted to note for future work that it could be nice to use `fqn.String` for these TRACE-type log messages, cause that'll then help us easily see which ones are legacy and which ones are not as we do future work, particularly if we end up needing to debug a panic from the `LegacyString` function. :grin: 

```
[TRACE] Fetching schema from provider "registry.terraform.io/-/aws"
```

Reply:
I absolutely agree, I just didn't want to add something that might startle users before we've started talking about it - I've been thinking that a blog post is in order!